### PR TITLE
[TASK] Add a lightweight development environment

### PR DESCRIPTION
**Pull request by KamiYang**
This patch adds a lightweight environment for development purposes.
To use/install it, use standard `composer install`. 

A TYPO3 instance will be prepared for 'FIRST_INSTALL'-steps in "/.Build/".
Docroot would be "/.Build/web/".

### CODE REVIEW

**Comment by KamiYang** [comment]
Blocked until https://review.typo3.org/#/c/58575/ is merged. After the patch, we won't need to explicit require "typo3/cms-recordlist".

File: composer.json [hunk 1]
```diff
@@ -8,14 +8,37 @@
...
+    "typo3/cms": {
+      "extension-key": "adminpanel_extended",
+      "cms-package-dir": "{$vendor-dir}/typo3/cms",
+      "app-dir": ".Build",
+      "web-dir": ".Build/web"
+    }
+  },
+  "scripts": {
+    "create-extension-directory": "mkdir -p .Build/web/typo3conf/ext",
+    "link-package": "ln -snvf ../../../../. .Build/web/typo3conf/ext/adminpanel_extended",
```
Comment (susannemoog) [inline composer.json]: nope, please don't use platform specific commands.

Comment (KamiYang) [reply composer.json]: Fair enough, but how would one then accomplish these things? Using a composer script class that is listening on events seems to be overkill.

Comment (KamiYang) [reply composer.json]: After thinking about it, a composer-script-event-class would be the solution. I'll introduce one later.

**Review by susannemoog — APPROVED** [review]
Looks better, thanks.

**Pull request merged**